### ISSUE TITLE
DNS Spoofing does not work

### ISSUE THREAD

**Issue by amateur02** [Ettercap/ettercap#889]
Hello, 
I have another issue und could again use some professional assistance.
I tried to do some DNS spoofing and it won’t work.
This is what I’ve done:
- I activated Apache and it is running.
![apache](https://user-images.githubusercontent.com/37872064/38513039-5237975e-3c2d-11e8-8d04-088fb1e73b4f.png)

- Starting Ettercap for en1
![hostlist](https://user-images.githubusercontent.com/37872064/38513094-77e901f4-3c2d-11e8-9e09-dd80d31d050f.png)

- Added router to Target 1 and my I-Pad to Target 2
![targets](https://user-images.githubusercontent.com/37872064/38513193-b6c34dc6-3c2d-11e8-9b0e-09f57e9a9d43.png)

- I started ARP poisoning
![arp poisoning](https://user-images.githubusercontent.com/37872064/38513206-c2510a16-3c2d-11e8-8399-04d7a225c99c.png)
![arp poisoning 2](https://user-images.githubusercontent.com/37872064/38513257-df915b58-3c2d-11e8-925e-04ea4b792d4b.png)

- I went to manage plugins and selected dns_spoofing
![dns spoofing 2](https://user-images.githubusercontent.com/37872064/38513975-1deeed64-3c30-11e8-8fcc-dae081434076.png)

Now when I’m trying to open microsoft.com on my I-Pad it should be redirected, according to the etter.dns config, to another page!?
![dns config](https://user-images.githubusercontent.com/37872064/38513337-33c882aa-3c2e-11e8-871e-816fb8552977.png)

All it shows is „Safari could not open the page because the server stopped responding“.
What changes do I have to make in the etter.dns config, to get requests (e.g. microsoft.com) from 
my I-Pad (192.168.2.100) redirected to my MacBook (192.162.2.101) to show the content 
of localhost. What am I missing?
Like already mentioned, I’m no professional, just learning. With one working example I will be 
able to get the rest. It would be good to see it working.
Your assistance will be highly appreciated!

**Comment by sgeto** [comment]
Hmmm... have you changed the IPs next to both microsoft.com A records entries to 192.162.2.101?

**Comment by nonokh** [comment]
@sgeto i have a similar issue.
i followed all the steps above and also configured my /etc/ettercap/etter.dns and kept my ip adresses accordingly.
![ip](https://user-images.githubusercontent.com/31509565/38542804-74fbb43a-3cc0-11e8-80ce-5fd443ad88e2.png)

Plus i am getting an L3 send error-unable to forward packets(permisson denied).
i am quite confused because i ran ettercap with _sudo ettercap -G_
![error](https://user-images.githubusercontent.com/31509565/38542871-a9f745c8-3cc0-11e8-9861-a6b9470dff6f.png)

**Comment by koeppea** [comment]
Make sure ettercap is build in debug mode. Then rerun and provide the ettercap_debug log file located in the execution directory.

Generally, famous sites like Microsoft.com or facebook.com are not the best to start testing in times of HSTS. Start with maybe your own.

Furthermore when ettercap spoofs a DNS reply, a message in the message pane appears. Haven’t seen this in any of the screenshots.

It important to know which IP a client uses for DNS resolution. E.g. when you only poison ARP but the client uses IPv6 for DNS resolution, ettercap won’t see the DNS query.

The operation not permitted may be something else, not related to dns_spoof.

**Comment by amateur02** [comment]
Hello,
thanks again for the swift response!
This may be a stupid question to you, like I’m said, I’m an amateur, but I have to ask. To redirect
someone within my privat network to my host, do I need to have an own domain?
Anyhow, I tried it again today with another web page and the end result was a little bit different.
Same setup as yesterday:
Localhost activated and running, victim was my I-Pad, and should have been redirected to my MacBook.
I adjusted the etter.dns to the other web page. Etter.dns config is attached to this message, so you
can also check if I messed something up or maybe you can run it on a computer to check.

- Router has been set as Target 1 and I-Pad as Target 2.

![target 1 und 2](https://user-images.githubusercontent.com/37872064/38565779-7720b978-3ce2-11e8-8abf-35539449b0a5.png)


- started ARP poisoning 

![arp](https://user-images.githubusercontent.com/37872064/38565796-809e9dee-3ce2-11e8-90fa-a0a8d85527aa.png)


- dns_spoof plugin activated

![activating spoof](https://user-images.githubusercontent.com/37872064/38565817-8edcde84-3ce2-11e8-9816-c8849e3b4cbf.png)


- On my I-Pad I typed the name of the web page requested. => this time I got some message in 
   the GUI, please see below, but on the I-Pad the page opened normal. Is this the message you
   are looking for?

![anzeige nach suche](https://user-images.githubusercontent.com/37872064/38565836-9cc30ffa-3ce2-11e8-80f1-893b814e9ec3.png)


As requested, I also attached the debug.log.
I hope it will help to find out why it is not working. I really appreciate your help solving this 
issue. 
[ettercap-0.8.2_debug.log](https://github.com/Ettercap/ettercap/files/1895205/ettercap-0.8.2_debug.log)

[etter.dns.zip](https://github.com/Ettercap/ettercap/files/1895233/etter.dns.zip)

**Comment by koeppea** [comment]
Honestly I begin to get confused.
The content of the provided _etter.dns.zip_ is not what is loaded (captured by _ettercap-0.8.2_debug.log_).
Ettercap expects by default (_and in your case_) the file `/etc/ettercap/etter.dns`.

For me it looks like, you've edited in the file in the **share** source directory past doing the `make install`.

**Comment by amateur02** [comment]
Okay, now I'm confused too. I didn't know that I have two different etter.dns on my 
MacBook.
Anyhow, this time I edited and modified etc/ettercap/etter.dns.

![etc ettercap etter_dns](https://user-images.githubusercontent.com/37872064/38823599-1bcfc95a-41a7-11e8-97b3-a169450b3503.png)


Like the other tests, I'm using my I-Pad as victim (192.168.2.100).

So I started Ettercap, selected my I-Pad as target 1 and my router as target 2.

![bildschirmfoto 2018-04-16 um 18 23 58](https://user-images.githubusercontent.com/37872064/38823628-29b700b0-41a7-11e8-927a-aaee802843a1.png)

Started arp poisoning

![bildschirmfoto 2018-04-16 um 18 24 30](https://user-images.githubusercontent.com/37872064/38823645-35b8cb00-41a7-11e8-83f9-ee0c53c544b8.png)

I've selected dns_spoofing at the plugin menu.

![bildschirmfoto 2018-04-16 um 18 25 00](https://user-images.githubusercontent.com/37872064/38823685-54895f2c-41a7-11e8-9b6c-00de00145836.png)

On my I-Pad I entered fynn-design.de and the page opened normally. No redirecting to my 
localhost, which was running.
As soon as the whole page was loaded I got the message below in the message pane, but nothing
else happened.

![bildschirmfoto 2018-04-16 um 18 29 10](https://user-images.githubusercontent.com/37872064/38823788-ad58ecf8-41a7-11e8-9b6b-b32c083207af.png)

You can see the complete message below.

HTTP : 82.198.79.63:80 -> USER: YTozOntzOjEyOiJjb250ZW50X3R5cGUiO3M6NDoicGFnZSI7czoxMDoiY29udGVudF9pZCI7aToyO3M6NjoiYXV0aG9yIjtzOjg6ImZpam5qYXJkIjt9.741007dc6c621a34becf0e582bdfe47e  PASS:   INFO: http://fynn-design.de/
CONTENT: action=slimtrack&op=add&id=YTozOntzOjEyOiJjb250ZW50X3R5cGUiO3M6NDoicGFnZSI7czoxMDoiY29udGVudF9pZCI7aToyO3M6NjoiYXV0aG9yIjtzOjg6ImZpam5qYXJkIjt9.741007dc6c621a34becf0e582bdfe47e&ref=&res=aHR0cDovL2Z5bm4tZGVzaWduLmRlLw==&sw=768&sh=1024&bw=768&bh=954&sl=1371&pp=127&pl=

Did you get a chance to check if you are able to spoof the above mentioned web page?
I also attached the debug.log from my last attempt and some pictures of the file structure how
ettercap was build at the installation.
Maybe you can see, what goes wrong. If I need to modify  or delete some corrupted parts,
let me know. 
I'm really happy with ettercap, but I also want to see dns spoofing working.

[ettercap-0.8.2_debug.log.zip](https://github.com/Ettercap/ettercap/files/1916394/ettercap-0.8.2_debug.log.zip)


[File structure.zip](https://github.com/Ettercap/ettercap/files/1916398/File.structure.zip)

**Comment by amateur02** [comment]
Sorry, got the wrong button!

**Comment by koeppea** [comment]
Reproduced using ettercap on MacOSX.
My victim is a Linux VM.

As expected I got the notification, that the DNS reply has been spoofed. As expected the website on the redirected web server is shown:
<img width="818" alt="ettercap-dns_spoof" src="https://user-images.githubusercontent.com/4201853/38830928-16604d52-41be-11e8-8bab-3b10498ade82.png">

The content of the _etter.dns_:
```
$ cat /etc/ettercap/etter.dns
fynn-design.de A 172.21.21.1
*.fynn-design.de A 172.21.21.1

$
```
I've selected my router as **TARGET1** and the victim as **TARGET2**.

So what could it be: 
 1. browser cache
 2. iPad doesn't accept the spoofed ARP packets, so that ettercap doesn't _see_ the DNS query
 3. iPad uses IPv6 for DNS, but you only poisons with ARP only effective for IPv4.



Can you try with another victim? Maybe it helps if you activate **arp_poisoning_smart** in `/etc/ettercap/etter.conf`.

**Comment by koeppea** [comment]
**arp_poisoning_request** in `/etc/ettercap/etter.dns` may also be worth a try.
See _man etter.conf_ for further information about the mentioned options.

**Comment by amateur02** [comment]
I followed your advice, except, I activated both arp_poisoning_smart and
arp_poisoning_request. 
Now it is working with both, my I-Pad (192.168.2.100) and my I-Phone (192.168.2.102) 
as you can see on the pictures. 

![i-phone](https://user-images.githubusercontent.com/37872064/38950560-3512d150-4346-11e8-8f43-359d08b25202.png)

![i-pad](https://user-images.githubusercontent.com/37872064/38950590-4b72d2c4-4346-11e8-9509-18ebbbcf9b62.png)


I didn't had enough time to really check which one did the trick, or if both
need to be activated.
I will try later and also with another victim.
Does it pose a problem to keep both activated? Do I have to expect other 
problems? Please let me know what you think.

**Comment by koeppea** [comment]
Coola banana. I originally added those two features 4 years  ago, just having in mind such a scenario. It goes down well that it's really helping in such situations, as they haven't been tested in wild (up to today).

The intention was, to make ettercap behave like a normal machine, but just with poisoned information. I suspected some modern devices introduce a anti-ARP spoof detection like apparently is the case meanwhile.

It's good to see it working. Thanks for testing. I'll discuss with the team, if and how and when we could change the default values making ettercap more robust against such counter measures. But most probably not for the upcoming release. For IPv6 (ND poisoning) I'll work on a similar approach.

Closing this issue now.
